cmd/pipelines-as-code-controller/main.go

```diff
@@ -28,9 +28,11 @@ func main() {
 		log.Fatal("failed to init kinit client : ", err)
 	}
 
-	if err := run.GetConfigFromConfigMap(ctx); err != nil {
-		log.Fatal("failed to get defaults : ", err)
-	}
+	go func() {
+		if err := run.WatchConfigMapChanges(ctx, run); err != nil {
```

Comment:
if configmap is not found we used to do `Fatal` 
now we are using go routine... ?
it will not fail right ? if configmap is missing

Reply:
or will it ? 😄

Reply:
It will if there are any error returning error will stop execution

I just added manual error here https://github.com/openshift-pipelines/pipelines-as-code/pull/785/files#diff-37150cc16df1329476c31a44285a213b2f8e2b82ce60813586589746b29aafd1R40 and observed that fatal works perfectly
```
kubectl logs -f pipelines-as-code-controller-6c75876df4-4f7hk -n pipelines-as-code 
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.801946228Z","caller":"logging/config.go:117","message":"Successfully created the logger."}
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.802196476Z","caller":"logging/config.go:118","message":"Logging level set to: info"}
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.80226619Z","caller":"logging/config.go:80","message":"Fetch GitHub commit ID from kodata failed","error":"open /var/run/ko/HEAD: no such file or directory"}
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.802395147Z","logger":"pipelinesascode","caller":"metrics/metrics_worker.go:76","message":"Flushing the existing exporter before setting up the new exporter."}
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.802891059Z","logger":"pipelinesascode","caller":"metrics/prometheus_exporter.go:51","message":"Created Prometheus exporter with config: &{pipelinesascode.tekton.dev/controller controller prometheus 5000000000 <nil>  false 9090 0.0.0.0}. Start the server for Prometheus exporter."}
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.80292176Z","logger":"pipelinesascode","caller":"metrics/metrics_worker.go:91","message":"Successfully updated the metrics exporter; old config: <nil>; new config &{pipelinesascode.tekton.dev/controller controller prometheus 5000000000 <nil>  false 9090 0.0.0.0}"}
{"severity":"INFO","timestamp":"2022-08-17T08:18:02.803294577Z","logger":"pipelinesascode","caller":"adapter/adapter.go:67","message":"Starting Pipelines as Code version: nightly"}
2022/08/17 08:18:02 unable to create watcher : manual error to verify
```